dbt/include/sqlserver/macros/adapters/relation.sql

```diff
@@ -1,5 +1,6 @@
 {% macro sqlserver__make_temp_relation(base_relation, suffix) %}
-    {% set tmp_identifier = '#' ~  base_relation.identifier ~ suffix %}
+    {% set uuid = run_query("SELECT REPLACE(NEWID(), '-','_') ")[0].values()[0] %}
```

Comment:
I'd prefer to generate this ID in Jinja instead of in SQL

Reply:
https://gist.github.com/hijonathan/5db3d8ec64e267d06c32 if someone wants to take a stab at it, if I have time I"ll come back to this

Reply:
How about
{%- set uuid = local_md5(base_relation.schema) -%}
?